host/resource/resource.go

```diff
@@ -0,0 +1,54 @@
+package resource
+
+import . "github.com/flynn/flynn/Godeps/_workspace/src/github.com/docker/docker/pkg/units"
+
+type Spec struct {
+	// Request is the amount of resource a job expects to consume, so the
+	// job should only be placed on a host with at least this amount of
+	// resource available, and once scheduled this amount of resource
+	// should then be unavailable on the given host.
+	Request int64 `json:"request"`
+
+	// Limit is an upper limit on the amount of resource a job can consume,
+	// the outcome of hitting this limit being implementation defined (e.g.
+	// a system error, throttling, catchable / uncatchable signals etc.)
+	Limit int64 `json:"limit"`
+}
+
+type Type string
+
+const (
+	TypeMemory   Type = "memory"
+	TypeMaxFD    Type = "max_fd"
```

Comment:
This limit is a wrapper around `RLIMIT_NOFILE` which limits the maximum file descriptor number, rather than the maximum number of file descriptors, hence the singular.

From the man page:

```
RLIMIT_NOFILE
	Specifies a value one greater than the maximum file descriptor number
	that can be opened by this process. Attempts (open(2), pipe(2), dup(2),
	etc.) to exceed this limit yield the error EMFILE.
```

For example:

```
$ ulimit -n
1024
$ echo 1023>/dev/null

$ echo 1024>/dev/null                                                     
bash: 1024: Bad file descriptor
$ ulimit -n 2048
$ echo 1024>/dev/null

$
```

Will add a comment.